### PR TITLE
[TRDT-308] Added version zero of the OpenAPI Spec.

### PR DESCRIPTION
**Pull request by erasmos-hmrc**

### CODE REVIEW

**Review by macgrewal — COMMENTED** [review]
Another question I have is related to how the data for the `/reports` endpoint is being passed. For consistency with some of the MTD ITSA APIs would it be worth including the `taxYear` and `calculationId` in the URI?

File: resources/public/api/conf/0.0.0/application.yaml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                  description: The ID (UUID) of a calculation.
+                customerType:
+                  type: string
+                  description: The type of customer.
+                  enum:
+                    - TaxPayer
+                    - Agent
+                taxYear:
+                  type: integer
+                  description: The tax year of the Self Assessment.
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply resources/public/api/conf/0.0.0/application.yaml]: Updated.

File: resources/public/api/conf/0.0.0/application.yaml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+              required:
+                - calculationId
+                - customerType
+                - taxYear
+              example:
+                calculationId: a85b059d-6b03-47d4-be97-60569925fbc9
+                customerType: TaxPayer
+                taxYear: 2000
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply resources/public/api/conf/0.0.0/application.yaml]: Updated.

**Comment by erasmos-hmrc** [comment]
> For consistency with some of the MTD ITSA APIs would it be worth including the taxYear and calculationId in the URI?

Interesting idea. 

But I'm not sure what I think about it. 

Anyway - I'll keep it like this for now.

**Review by macgrewal — APPROVED** [review]

**Review by Krishnarepogit — APPROVED** [review]
As Mac reviewed this ...

**Pull request merged**